pkg/function/client/client.go

```diff
@@ -0,0 +1,88 @@
+package client
+
+import (
+	"context"
+	"fmt"
+
+	functionpb "github.com/numaproj/numaflow-go/pkg/apis/proto/function/v1"
+	"github.com/numaproj/numaflow-go/pkg/function"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+// Client contains the grpc connection and the grpc client.
+type Client struct {
+	conn    *grpc.ClientConn
+	grpcClt functionpb.UserDefinedFunctionClient
+}
+
+// New creates a new client object.
+func New(inputOptions ...Option) (*Client, error) {
+	var opts = &options{
+		mockClient: nil,
```

Comment:
This is an interesting approach to get mock integrated, do you have an example that any other projects are doing this way? The brings some concern to me, that a mockclient becomes an official option of calling this API, which is supposed to be an internal stuff for testing.

Reply:
> do you have an example that any other projects are doing this way?

No.. it's just my ideas....

The numaflow imports the client and needs to have a way to get a mock client for testing. 
client.New() is called here for real use case
https://github.com/chromevoid/numaflow/blob/udf-grpc-client/pkg/udf/applier/uds_grpc.go#L23-L29
and here for tests
https://github.com/chromevoid/numaflow/blob/udf-grpc-client/pkg/udf/applier/uds_grpc_test.go#L22-L25

Reply:
we shouldn't leak mocking to the official option.